### PR TITLE
ci: add make_unique rule to check_format

### PR DESCRIPTION
**Pull request by silentdai**
Signed-off-by: Yuchen Dai <silentdai@gmail.com>

As a follow up of #7034 
```
$ ./tools/check_format.py  check tools/testdata/check_format/ |grep cpp_std |grep absl
ERROR: tools/testdata/check_format//cpp_std.cc:7: cpp std: absl::make_unique should be replaced by std::make_unique
```

tag #7053

### CODE REVIEW

**Comment by silentdai** [comment]
Nice. It works. Maybe ci rebases from master? I don't see the miss usage at the current commit. Will rebase and fix.
```
ERROR: From ./source/server/server.cc
ERROR: ./source/server/server.cc:553: cpp std: absl::make_unique should be replaced by std::make_unique
ERROR: ./source/server/server.cc:561: cpp std: absl::make_unique should be replaced by std::make_unique
ERROR: From ./source/extensions/quic_listeners/quiche/platform/quic_ptr_util_impl.h
ERROR: ./source/extensions/quic_listeners/quiche/platform/quic_ptr_util_impl.h:17: cpp std: absl::make_unique should be replaced by std::make_unique
ERROR: From ./source/common/network/addr_family_aware_socket_option_impl.h
ERROR: ./source/common/network/addr_family_aware_socket_option_impl.h:22: cpp std: absl::make_unique should be replaced by std::make_unique
ERROR: ./source/common/network/addr_family_aware_socket_option_impl.h:23: cpp std: absl::make_unique should be replaced by std::make_unique
ERROR: From ./source/common/router/rds_impl.cc
ERROR: ./source/common/router/rds_impl.cc:219: cpp std: absl::make_unique should be replaced by std::make_unique
ERROR: From ./test/common/http/codec_impl_fuzz_test.cc
ERROR: ./test/common/http/codec_impl_fuzz_test.cc:358: cpp std: absl::make_unique should be replaced by std::make_unique
ERROR: ./test/common/http/codec_impl_fuzz_test.cc:362: cpp std: absl::make_unique should be replaced by std::make_unique
ERROR: ./test/common/http/codec_impl_fuzz_test.cc:369: cpp std: absl::make_unique should be replaced by std::make_unique
ERROR: ./test/common/http/codec_impl_fuzz_test.cc:374: cpp std: absl::make_unique should be replaced by std::make_unique
ERROR: check format failed. run 'tools/check_format.py fix'
```

File: tools/check_format.py [hunk 1]
```diff
@@ -75,6 +75,9 @@
     "ProtobufWkt::MapPair":             "Protobuf::MapPair",
     "ProtobufUtil::MessageDifferencer": "Protobuf::util::MessageDifferencer"
 }
+CPP_STD_ERRORS = {
+    "absl::make_unique": "std::make_unique",
```
Comment (jmarantz) [inline tools/check_format.py]: do you want to put the "<" into both the 'from' and 'to' mappings here to make it less likely to mess with comments?

File: tools/testdata/check_format/cpp_std.cc [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#include <memory>
+
+#include "absl/memory/memory.h"
+
+namespace Envoy {
+
+std::unique_ptr<int> bad = absl::make_unique<int>(0);
```
Comment (jmarantz) [inline tools/testdata/check_format/cpp_std.cc]: thanks for adding the test data. But you'll need to also add some lines to tools/check_format_test_helper.py to run on this file with expected outputs. Scan to the end of that file and you'll see what to do.

File: tools/check_format.py (hunk 1 above)
Comment (silentdai) [reply tools/check_format.py]: Good point! Fixed

File: tools/testdata/check_format/cpp_std.cc (hunk 2 above)
Comment (silentdai) [reply tools/testdata/check_format/cpp_std.cc]: Thanks! Added

File: tools/check_format.py [hunk 3]
```diff
@@ -370,6 +377,9 @@ def checkSourceLine(line, file_path, reportError):
     if invalid_construct in line:
       reportError("incorrect protobuf type reference %s; "
                   "should be %s" % (invalid_construct, valid_construct))
+  for invalid_construct, valid_construct in CPP_STD_ERRORS.items():
+    if invalid_construct in line:
+      reportError("cpp std: %s should be replaced by %s" % (invalid_construct, valid_construct))
```
Comment (jmarantz) [inline tools/check_format.py]: Can we phrase this:

 function %s should be replaced with standard library function %s

and give the entire elaborated version of that as the expected error message in check_format_test_helper.py.

File: tools/check_format.py [hunk 4]
```diff
@@ -75,6 +75,9 @@
     "ProtobufWkt::MapPair":             "Protobuf::MapPair",
     "ProtobufUtil::MessageDifferencer": "Protobuf::util::MessageDifferencer"
 }
+CPP_STD_ERRORS = {
```
[the remark this answers is not in the document]
Comment (silentdai) [reply tools/check_format.py]: done

File: tools/check_format.py (hunk 3 above)
Comment (silentdai) [reply tools/check_format.py]: Fixed except for word "function". I believe there will be type as well.

**Comment by jmarantz** [comment]
LGTM generally.

Looks like another reference to absl::make_unique got merged into ./test/mocks/upstream/cluster_info.cc; let's fix that one and get this merged.

**Comment by silentdai** [comment]
Thanks! Looks like this PR has greater value than I expected 
Rebase and force pushed

**Comment by silentdai** [comment]
oops. looking into the failure

File: tools/testdata/check_format/cpp_std.cc [hunk 5]
```diff
@@ -0,0 +1,11 @@
+#include <memory>
+
+#include "absl/memory/memory.h"
```
Comment (jmarantz) [inline tools/testdata/check_format/cpp_std.cc]: btw this file (and the gold) is not actually compiled, so you don't need the includes. You do need the 'namespace Envoy' as the presence of that is one of the formatting rules that is checked.

Comment (jmarantz) [reply tools/testdata/check_format/cpp_std.cc]: Cleaning those out is totally optional IMO.

**Pull request merged**